src/posix_disk_io.cpp

```diff
@@ -186,7 +186,7 @@ namespace {
 			posix_storage* st = m_torrents[storage].get();
 
 			int const piece_size = st->files().piece_size(piece);
-			int const blocks_in_piece = (piece_size + default_block_size - 1) / default_block_size;
+			int const blocks_in_piece =  st->files().blocks_in_piece2(piece);
```

Comment:
this was an old bug. I'm tempted to put this in master as a separate commit too, ahead of v2

Reply:
`blocks_in_piece2` should only be used to compute v2 hashes, the v1 hash needs to use the v1 piece size. This function should do what `disk_io_thread::do_hash` does.

Reply:
the problem was(/is) that `block_hashes` past in here has size 64, but the previous way of computing `blocks_in_pieces` was 128, meaning we overran the end of the span in the loop below. Perhaps this is not the right fix, I copied the logic from the call site that allocates the storage for the span